src/pkg/services/m365/api/drive_test.go

```diff
@@ -51,3 +72,195 @@ func (suite *OneDriveAPISuite) TestCreatePagerAndGetPage() {
 	assert.NoError(t, err, clues.ToCore(err))
 	assert.NotNil(t, a)
 }
+
+// newItem initializes a `models.DriveItemable` that can be used as input to `createItem`
+func newItem(name string, folder bool) *models.DriveItem {
+	itemToCreate := models.NewDriveItem()
+	itemToCreate.SetName(&name)
+
+	if folder {
+		itemToCreate.SetFolder(models.NewFolder())
+	} else {
+		itemToCreate.SetFile(models.NewFile())
+	}
+
+	return itemToCreate
+}
+
+func (suite *DriveAPISuite) TestDrives_PostItemInContainer() {
+	t := suite.T()
+
+	ctx, flush := tester.NewContext(t)
+	defer flush()
+
+	rc := tester.DefaultTestRestoreConfig("drive_api_post_item")
+
+	// generate a parent for the test data
+	parent, err := suite.ac.Drives().PostItemInContainer(
+		ctx,
+		suite.driveID,
+		suite.rootFolderID,
+		newItem(rc.Location, true),
+		control.Replace)
+	require.NoError(t, err, clues.ToCore(err))
+
+	// generate a folder to use for collision testing
+	folder := newItem("collision", true)
+	origFolder, err := suite.ac.Drives().PostItemInContainer(
+		ctx,
+		suite.driveID,
+		ptr.Val(parent.GetId()),
+		folder,
+		control.Copy)
+	require.NoError(t, err, clues.ToCore(err))
+
+	// generate an item to use for collision testing
+	file := newItem("collision.txt", false)
+	origFile, err := suite.ac.Drives().PostItemInContainer(
+		ctx,
+		suite.driveID,
+		ptr.Val(parent.GetId()),
+		file,
+		control.Copy)
+	require.NoError(t, err, clues.ToCore(err))
+
+	mus, err := suite.ac.Drives().NewItemContentUpload(
+		ctx,
+		suite.driveID,
+		ptr.Val(origFile.GetId()))
+	require.NoError(t, err, clues.ToCore(err))
+
+	iw := graph.NewLargeItemWriter(
+		ptr.Val(origFile.GetId()),
+		ptr.Val(mus.GetUploadUrl()),
+		int64(len("{}")))
+
+	_, err = iw.Write([]byte("{}"))
+	require.NoError(t, err, clues.ToCore(err))
+
+	table := []struct {
+		name        string
+		onCollision control.CollisionPolicy
+		postItem    models.DriveItemable
+		expectErr   func(t *testing.T, err error)
+		expectItem  func(t *testing.T, i models.DriveItemable)
+	}{
+		{
+			name:        "fail folder",
+			onCollision: control.Skip,
+			postItem:    folder,
+			expectErr: func(t *testing.T, err error) {
+				assert.ErrorIs(t, err, graph.ErrItemAlreadyExistsConflict, clues.ToCore(err))
+			},
+			expectItem: func(t *testing.T, i models.DriveItemable) {
+				assert.Nil(t, i)
+			},
+		},
+		{
+			name:        "rename folder",
+			onCollision: control.Copy,
+			postItem:    folder,
+			expectErr: func(t *testing.T, err error) {
+				assert.NoError(t, err, clues.ToCore(err))
+			},
+			expectItem: func(t *testing.T, i models.DriveItemable) {
+				assert.NotEqual(
+					t,
+					ptr.Val(origFolder.GetId()),
+					ptr.Val(i.GetId()),
+					"replaced item should be different from the original")
```

Comment:
Should also check the names aren't the same, after all that's the change the user will see, not the ID change
```suggestion
					"renamed item should be different from the original")
				assert.NotEqual(
					t,
					ptr.Val(origFolder.GetName()),
					ptr.Val(i.GetName()),
					"renamed item should have different name")
```